### ISSUE TITLE
Make loading default stacks more obvious

### ISSUE THREAD

**Issue by ashumilova** [eclipse-che/che#3629]
Now the stacks.json rewrites the existing stacks each time the user restarts che server. In this case - user makes his changes to existing stacks, restarts che and looses the changes. It should be better and more clear for user to load stacks.json only on the first che start  - to fill database with predefined values.

**Comment by skabashnyuk** [comment]
We are proposing following concept: 
 - remove stacks.json and template.json
 - Move stacks and template initialization to database script

 That makes two different strategies for two different group of users.

1. Codenvy and Che users if they want to customize their installation they need to go to dashboard and remove or add or modify any stack they want.
   If they want to copy some stack from one installation to another they just need to copy\paste it on dashboard.

2. Users who is making new assembly of Codenvy and Che they need to extend database initialization script. Remove everything that is not needed
add new stacks and templates. Can be simplified by creating stack from dashboard and some tooling with database.

**Comment by TylerJewell** [comment]
So there are two types of issues:
1. When we have a custom assembly, and admins initialized their database with a default set of stacks / templates, and then users update it within the dashboard, those user changes are overwritten.  We should avoid this.

2. We would need some details on the custom assembly scenario on what it means to extend the database initialization script.  This sounds complicated, even though it is just a text file.  So, please keep in mind that someone who runs an archetype to get a custom assembly will need a simple way to find where stacks / templates are loaded by database, edit it, and then debug it if the format they type is wrong.

**Comment by skabashnyuk** [comment]
>? those user changes are overwritten. We should avoid this.
 
That is not going to happen when move stacks.json and template.json to database initialization script

>We would need some details on the custom assembly scenario
Yes sure.  We can simplify it with some tooling like
   - cli export stacks > stack.sql
   - export to file in UD

**Comment by garagatyi** [comment]
It was very confusing when I found that if user started using Che from 5.x.0 and updated his Che after each release 5.x+1.0, 5.x+2.0, ..., 5.x+n.0 he won't get updates in old stacks from these releases.
So looking at [page with changes in stacks](https://github.com/eclipse/che/commits/master/ide/che-core-ide-stacks/src/main/resources/stacks.json) note that changes made in this file is never delevered to a user that started using Che before that release. Except addition of new stacks, they are added into Che.

Source of this behavior is that CLI copies stacks.json file on che start if it [doesn't exist](https://github.com/garagatyi/che/blob/798ca081b4e3f72d38afddee525f5e3bdd07e6bb/dockerfiles/che/entrypoint.sh#L289-L289).
But because of that user won't get updates of the stacks that were in the file when it was copied.
Moreover this file is supposed to help users that add their own stacks but even if user has never edited that file he won't get updates because of the described behavior.

I merged separation of terminal and exec agents in this release and when we remove hack that adds exec into each machine with terminal(probably 5.8.0) mysql stack will be broken.
